### ISSUE TITLE
Bloks panel stay open

### ISSUE THREAD

**Issue by EddyBoels** [WordPress/gutenberg#38877]
### Description

Bloks/reusable/patterns panel cant be close after visiting explorer.

### Step-by-step reproduction instructions

Open bloks panel
Go to patterns tab
Clik explorer (don't chose any pattern)
Close Explorer
Try to close panel

### Screenshots, screen recording, code snippet

![patternspanel](https://user-images.githubusercontent.com/99866057/154426597-d557d72f-c6b8-4e21-b9fc-c1ec5bae357a.gif)

### Environment info

WordPress 5.9
Tested with Twenty Twenty-Two and Twenty Twenty-One

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by talldan** [comment]
I think there's a deeper bug here. There's a focus loss when closing the Explore modal. It should be transferred back to the 'Explore' button. Doing so would also fix the issue with the sidebar not closing (it seems to only happen if focus is programatically transferred outside the sidebar).

You can see that when the explore modal is open, the sidebar in the background has no contents (if your browser window is wide enough), so this is no doubt why focus isn't transferred back—there's no element to return focus to.
